crates/build/src/lib.rs

```diff
@@ -161,6 +161,53 @@ where
     (sources, validations)
 }
 
+/// The output of a build, which can be either successful, failed, or anything
+/// in between. The "in between" may seem silly, but may be important for
+/// some use cases. For example, you may be executing a build for the purpose
+/// of getting the collection projections, in which case you may not want to
+/// consider errors from materialization validations to be build failures.
+pub struct BuildOutput {
```

Comment:
nit: should we call this Output so it doesn't stutter when used as `build::Output` ?

Reply:
`build::Build` could also work